Firestore/core/src/api/firestore.h

```diff
@@ -113,6 +115,8 @@ class Firestore : public std::enable_shared_from_this<Firestore> {
   std::shared_ptr<util::Executor> user_executor_;
   std::shared_ptr<util::AsyncQueue> worker_queue_;
 
+  std::unique_ptr<remote::FirebasePlatformLogging> firebase_platform_logging_;
```

Comment:
This should be declared before `FirestoreClient` so that it outlives that instance during destruction. Indeed, it seems like `FirestoreClient` should be among the last things declared in here.